akka-stream/src/main/scala/akka/stream/impl/io/TcpListenStreamActor.scala

```diff
@@ -151,10 +155,17 @@ private[akka] class TcpListenStreamActor(localAddressPromise: Promise[InetSocket
     val (connected: Connected, connection: ActorRef) = incomingConnections.dequeueInputElement()
     val tcpStreamActor = context.watch(context.actorOf(TcpStreamActor.inboundProps(connection, halfClose, settings)))
     val processor = ActorProcessor[ByteString, ByteString](tcpStreamActor)
+
+    import scala.concurrent.duration._
+    val handler = (idleTimeout match {
+      case d: FiniteDuration ⇒ Flow[ByteString].join(Timeouts.idleTimeoutBidi[ByteString, ByteString](d))
+      case _                 ⇒ Flow[ByteString]
+    }).andThenMat(() ⇒ (processor, ()))
```

Comment:
I am not sure what is expected here. This is an *incoming* TCP connection, so there is no way you can materialize it twice, since the connection is created by the client. We discussed with @rkuhn to make it reusable by implicitly adding merge and broadcast logics to the input/output sides of the connection, but that in my opinion is much worse as it can lead to silent data corruption on the wire (if someone is not aware of the merging) while the current approach throws an exception instead.